spyder/widgets/variableexplorer/utils.py

```diff
@@ -174,6 +174,14 @@ def datestr_to_datetime(value):
     return v
 
 
+def str_to_timedelta(value):
+    m = re.match('^(?:datetime\.timedelta)?\(?([^\)]*)\)?$', value)

```

Comment:
What's this regexp trying to match? Could you post an example, please?

Reply:
The regexp matches `datetime.timedelta(ANYTHING)` where
`datetime.timedelta` and the brackets are optional. ANYTHING is then further processed.

I assume that ANYTHING is either a single `int` or a list of `ints` separated by colons and possibly whitespace. Examples of matches:

- `datetime.timedelta(1)`
- `datetime.timedelta(1, 5, 12345)`
- `(1, 5, 12345)`
- `1`
- `1, 5, 12345`

all are converted to `datetime.timedelta(1)` or `datetime.timedelta(1, 5, 12345)` obejcts.

These represent the valid string inputs if the user wants to set a new value to a `datetime.timedelta` object via the variable explorer.

Reply:
Ok, thanks for the explanation. It's clear to me now :+1:

Reply:
Should then `timedelta(1)` be valid as well?

Reply:
> Should then timedelta(1) be valid as well?

Fine by me. Any reasonable string could be included. We can simply adapt the regexp.

> Feels like we should probably have a date/delta editor that has a box for each part?

Wouldn't be wrong. But for me, that's too much effort. I won't implement it. After all, editing timedeltas is probably a niche case and a string interface is good enough.

Reply:
Yeah, I agree.

Reply:
I've extended the matching to `timedelta(1)` and added some more documentation

Reply:
Thanks :-)